### PR TITLE
🪟 🎉 Adds status change icon for connections

### LINKED ISSUE #17829: Show auto-detect schema changes status in connection list view
Show that changes were detected on the connections list.

Prototype:
https://www.figma.com/proto/liLQhcbpVHiEDW18kiXQe3/01_03_CONNECTIONS?node-id=857%3A25233&scaling=min-zoom&page-id=857%3A24319&starting-point-node-id=857%3A25233&show-proto-sidebar=1

### PR DESCRIPTION
**Pull request by YatsukBogdan1**
## What
Closes #17829

## How
Adds Icon for `<StatuCell />` to represent `schemaChange` value on UI

Screenshot 2022-11-23 at 16 46 06
Screenshot 2022-11-23 at 16 46 34

### CODE REVIEW

File: airbyte-webapp/src/locales/en.json
Comment (timroes) [inline airbyte-webapp/src/locales/en.json:394]: I think we should work on the wording here a bit. Those errors are currently very "non helpful" in a way that they are very technical and don't tell the user what's actually asked from them to do here. We should try to phrase them in a way that makes clear what this warnings/errors are meaning for their data as well as what we expect them to do to solve it.

@andyjih can maybe help with this.

File: airbyte-webapp/src/components/EntityTable/components/StatusCellControl.tsx
Comment (dizel852) [inline airbyte-webapp/src/components/EntityTable/components/StatusCellControl.tsx:53]: @krishnaglick how do think, should we show the "status change" if sync is in progress? 🤔

File: airbyte-webapp/src/components/EntityTable/components/StatusCellControl.tsx
Comment (dizel852) [reply airbyte-webapp/src/components/EntityTable/components/StatusCellControl.tsx:30]: Hm.. curious, it wasn't added previously 🤔

**Review by dizel852 — CHANGES_REQUESTED** [review]
Code LGTM, also tested locally.
Need to fix PR comments from other reviewers.

@andyjih any updates on i18n translations?

File: airbyte-webapp/src/locales/en.json
Comment (edmundito) [reply airbyte-webapp/src/locales/en.json:394]: We can reuse the wording in #19226 for these. Also, breaking does not mean that the protocol has changed. That is a different feature that has not been implemented in the frontend yet.

File: airbyte-webapp/src/components/EntityTable/components/StatusCellControl.tsx
Comment (edmundito) [inline airbyte-webapp/src/components/EntityTable/components/StatusCellControl.tsx:24]: I would recommend moving this back to StatusCell and just adding a small wrapper for the div at the bottom of the file. Otherwise, this will cause a lot of conflicts with #19226, which will also add tests for the component

File: airbyte-webapp/src/components/EntityTable/components/StatusCell.tsx
Comment (edmundito) [inline airbyte-webapp/src/components/EntityTable/components/StatusCell.tsx:30]: Need to check if the feature is enabled using the current flagging, for example:

```tsx
const isSchemaChangesFeatureEnabled = process.env.REACT_APP_AUTO_DETECT_SCHEMA_CHANGES = "true";

...
{isSchemaChangesFeatureEnabled && <ChangeStatusIcon ... />}
```

**Review by edmundito — CHANGES_REQUESTED** [review]

File: airbyte-webapp/src/components/EntityTable/components/StatusCell.tsx
Comment (krishnaglick) [inline airbyte-webapp/src/components/EntityTable/components/StatusCell.tsx:67]: Rather than re-assign ControlComponent, I think it would be good to turn it into its own component and conditionally return the right JSX in there.

File: airbyte-webapp/src/components/EntityTable/components/StatusCell.tsx
Comment (krishnaglick) [inline airbyte-webapp/src/components/EntityTable/components/StatusCell.tsx:31]: This can be hoisted outside the component since it doesn't change.

Comment (edmundito) [reply airbyte-webapp/src/components/EntityTable/components/StatusCell.tsx:31]: It's fine to leave it here because this will soon change to use the experiment hook

**Review by dizel852 — CHANGES_REQUESTED** [review]
@YatsukBogdan1 
Some comments still need to be fixed.
Also, please resolve the fixed ones.

File: airbyte-webapp/src/components/EntityTable/components/StatusCell.tsx
Comment (YatsukBogdan1) [reply airbyte-webapp/src/components/EntityTable/components/StatusCell.tsx:67]: the initial idea was the same, but @edmundito suggest to keep everything in one component to make merging with https://github.com/airbytehq/airbyte/pull/19226 easier. after this PR will be merged, I am planing to return to my initial idea with separate component

Comment (dizel852) [reply airbyte-webapp/src/components/EntityTable/components/StatusCell.tsx:67]: Oh, yeah, sure. We also need to take into account that. 👍

**Review by dizel852 — APPROVED** [review]
@krishnaglick waiting for your reply on comment - https://github.com/airbytehq/airbyte/pull/19755/files#r1037255876

Meanwhile - Approved.
Passing the final review to @edmundito and @krishnaglick

**Comment by andyjih** [comment]
> Code LGTM, also tested locally. Need to fix PR comments from other reviewers.
> 
> @andyjih any updates on i18n translations?

Do we do i18n translations? This is the first time I've heard of them. What do you need from me?

**Comment by dizel852** [comment]
> > Code LGTM, also tested locally. Need to fix PR comments from other reviewers.
> > @andyjih any updates on i18n translations?
> 
> Do we do i18n translations? This is the first time I've heard of them. What do you need from me?

Actually, all translations that we have in `en.json` file they are all i18n translations 😄 Sorry to confuse you.

We need your opinion regarding this comment - https://github.com/airbytehq/airbyte/pull/19755#discussion_r1030538240
Can we use the wording that @edmundito suggested? 
I personally second his suggestion 🙂

**Comment by edmundito** [comment]
@YatsukBogdan1 Merged #19226 - You should now be able to resolve the conflicts with StatusCell and then use the strings in en.json for the status message.

File: airbyte-webapp/src/components/EntityTable/components/StatusCell.tsx
Comment (YatsukBogdan1) [reply airbyte-webapp/src/components/EntityTable/components/StatusCell.tsx]: fixed this, since @edmundito was merged

**Comment by YatsukBogdan1** [comment]
> @YatsukBogdan1 Merged #19226 - You should now be able to resolve the conflicts with StatusCell and then use the strings in en.json for the status message.

done. I returned my initial wrapper component and put logic into `StatusCellControl` and also fixed localization

File: airbyte-webapp/src/locales/en.json
Comment (andyjih) [reply airbyte-webapp/src/locales/en.json]: +1 to @edmundito on using the same wording there

**Review by dizel852 — APPROVED** [review]
Tested one more time after fixes - seems all okay.
Passing to @edmundito

File: airbyte-webapp/src/components/EntityTable/components/StatusCellControl.tsx
Comment (krishnaglick) [reply airbyte-webapp/src/components/EntityTable/components/StatusCellControl.tsx:70]: I'm not sure I understand this question.

File: airbyte-webapp/src/components/EntityTable/components/ChangesStatusIcon.tsx
Comment (krishnaglick) [inline airbyte-webapp/src/components/EntityTable/components/ChangesStatusIcon.tsx:40]: This tooltip bit can be cut way down.
```
<FormattedMessage id={`connection.schemaChange.${schemaChange === "non_breaking" ? "nonBreaking" : schemaChange}`} />
```
Or even just run the `schemaChange` through https://stackoverflow.com/a/61375162

Comment (YatsukBogdan1) [reply airbyte-webapp/src/components/EntityTable/components/ChangesStatusIcon.tsx]: thanks for the utility function idea. added to preject

**Review by krishnaglick — APPROVED** [review]

File: airbyte-webapp/src/components/EntityTable/components/ChangesStatusIcon.tsx
Comment (edmundito) [inline airbyte-webapp/src/components/EntityTable/components/ChangesStatusIcon.tsx:29]: Add `placement="left"` Otherwise, the tooltip doesn't look great for breaking changes:
image


With `placement="left"`:
image

File: airbyte-webapp/src/components/EntityTable/components/StatusCell.tsx
Comment (edmundito) [inline airbyte-webapp/src/components/EntityTable/components/StatusCell.tsx:30]: The environment variable is `process.env.REACT_APP_AUTO_DETECT_SCHEMA_CHANGES`

**Review by edmundito — CHANGES_REQUESTED** [review]
Tested. The implementation looks good but there were a couple of issues that I noticed.

**Review by edmundito — APPROVED** [review]

**Pull request merged**
